lib/api/slack.ex

```diff
@@ -16,12 +18,44 @@ defmodule Aprb.Api.Slack do
       requires :response_url, type: String
     end
     post do
+      # check that token matches, that the POST comes from our slack integration
+      if System.get_env("SLACK_SLASH_COMMAND_TOKEN") != params[:token] do
+        conn
+          |> put_status(403)
+          |> json(%{message: "Unauthorized"})
+          |> halt()
+      end
 
-      # TODO: check that token matches, that the POST comes from our slack integration
-      # if System.get_env("SLACK_SLASH_COMMAND_TOKEN") != params["token"] ...
+      subscriber = case Repo.get_by(Subscriber, user_id: params[:user_id]) do
+          nil ->
+            # create new subscriber
+            sub_changeset = Subscriber.changeset(%Subscriber{}, Map.take(params, [:team_id, :team_domain, :channel_id, :channel_name, :user_id, :user_name]))
+            case Repo.insert(sub_changeset) do
+              {:ok, new_subscriber} ->
+                subscriber = new_subscriber
+              {:error, _changeset} ->
+                json(conn, %{ message: "Can't create user"})
+            end
+          existing_subscriber ->
+            existing_subscriber
+        end
 
-      IO.inspect params
-      json(conn, %{ hello: :world })
+      response = cond do
+        params[:text] == "list" ->
+          Repo.all(
+            from topics in Topic,
+            select: topics.name)
+
+        params[:text] == "my-subs" ->
+          Repo.preload(subscriber, :topics).topics
+
+        Regex.match?( ~r/subscribe/ , params[:text])  ->
+          # add subscriptions
+          IO.inspect params[:text]
+        true ->
+          "Unknown command! Supported commands: list, my-subs, subscribe <list of topics>"
```

Comment:
I think `topic` is a better name for `list`, so I would do `apr topics`, `apr subscriptions` and `apr subscribe <topic>`.